var/spack/repos/builtin/packages/octave/package.py

```diff
@@ -88,12 +88,13 @@ class Octave(AutotoolsPackage, GNUMirrorPackage):
     depends_on('gnuplot',      when='+gnuplot')
     depends_on('imagemagick',  when='+magick')
     depends_on('hdf5',         when='+hdf5')
-    depends_on('java',          when='+jdk')        # TODO: requires Java 6 ?
+    depends_on('java',         when='+jdk')        # TODO: requires Java 6 ?
     depends_on('llvm',         when='+llvm')
-    # depends_on('opengl',      when='+opengl')    # TODO: add package
+    depends_on('gl',           when='+opengl')
```

Comment:
What's the difference between `gl` here and `opengl` below?

Reply:
I just saw many other package depend on 'gl' and none on 'opengl'. But this is not an explanation, sorry.

Reply:
I think it may be that `gl` is a package, but `opengl` in `qt+opengl` is a build option.

Maybe the fltk package should also support a `+opengl` option?

Reply:
Exactly for both. But maybe only.

Reply:
`gl` is a virtual dependency (like e.g. `mpi` or `lapack`) with many implementations:
```console
$ spack providers gl
gl:
mesa    opengl       opengl@1.1:  opengl@1.3:  opengl@1.5:  opengl@2.1:  opengl@3.1:  opengl@3.3:  opengl@4.2:  opengl@4.4:
mesa18  opengl@1.0:  opengl@1.2:  opengl@1.4:  opengl@2.0:  opengl@3.0:  opengl@3.2:  opengl@4.1:  opengl@4.3:  opengl@4.5:
```
If a package just depends on `gl` it depends on any other package providing the API or functionality. If it depends on `opengl` it depends on that specific implementation (as if it dependend on `openblas` or `mpich`). There are also variants with the same name to activate dependencies in case it makes sense to have them conditional, but that's a separate topic.